File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        if (lists.length == 0) return null;
        for (int i = 1; i < lists.length; i++) {
            lists[i] = mergeTwoLists(lists[i], lists[i-1]);
        }

        return lists[lists.length-1];

    }

    public ListNode mergeTwoLists(ListNode one, ListNode two) {
        ListNode head = null;
        if (one.val < two.val) {
            head = one;
            one = one.next;
        } else {
            head = two;
            two = two.next;
        }
        ListNode current = head;

        while (one != null && two != null) {
            if (one.val < two.val) {
                current.next = one;
                one = one.next;
            } else {
                current.next = two;
                two = two.next;
            }
            current = current.next;
        }

        if (one == null) current.next = two;
        if (two == null) current.next = one;

        return head;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        cur = head 
        len = 0
        while cur:
            cur = cur.next
            len += 1
        
        print(len)
        if n == len:
            return head.next

        m = len - n
        cur = head
        while m > 1:
            m -= 1
            cur = cur.next

        cur.next = cur.next.next
        return head
            